content/engineering/guest-post/multi-factor-authentication-using-redis-cache-and-otp/index.md

```diff
@@ -0,0 +1,628 @@
+1. Navigate to [Outlook’s sign up page](https://outlook.live.com/owa/?nlp=1&signup=1) to register for an Outlook account if you don’t have one.
+
+2. Add your phone number if you are prompted to do so. 
+    
+    > This step can be skipped.
+
+### Sending Emails
+
+After setting up your Outlook account, the next step is to use your Outlook account’s credential to send emails through Nodemailer.
+
+1. Open your terminal and install the `nodejs-nodemailer-outlook` dependency.
+
+    This module only needs your Outlook account credentials and your message set up to send emails.
+
+    Now, run the code below in your terminal to install the `nodejs-nodemailer-outlook` dependency:
+    
+    ```bash
+    npm i nodejs-nodemailer-outlook
+    ```
+    
+2. Import the `nodejs-nodemailer-outlook` module into `server.js` and add the configurations needed to send emails:
+    
+    ```js
+    const express = require("express");
+    const app = express();
+    app.use(express.json());//Enabling the body attribute.
+    const PORT = 5000;
+    
+    app.post("/auth", (req, res) => {//Post request to 'http://localhost:3000'
+      const nodeoutlook = require("nodejs-nodemailer-outlook");//Importing the module
+      nodeoutlook.sendEmail({
+        auth: {//Outlook Account Credentials 
+          user: "<yourOutlookEmail@outlook.com>",//Email
+          pass: "<yourOutlookPassword>",//Password
+        },
+        from: "<yourOutlookEmail@outlook.com>",
+        to: req.body.email,//Getting the email
+        subject: "OTP",
+        text: req.body.message,//Getting the Message
+        onError: (e) => res.status(500).json({ message: `an error occurred ${e}` }),
+        onSuccess: (i) => res.status(200).json({ message: i }),
+      });
+    });
+    
+    app.listen(PORT, () => {
+      console.log(`Server running on port ${PORT}`);
+    });
+    ```
+
+3. To test the application, use an API testing application like [Postman](https://www.postman.com/) and make a POST request to http://localhost:5000/auth, passing `email` and `message` to the body of the request.
+
+    ![Postman](postman.jpg)
+
+## Generate OTPs
+
+In the previous section, you've successfully configured Nodemailer and used it to send emails. This section will show you how to generate a random number and how to add this random number to the message sent to the user.
+
+You can generate this random number using: 
+
+1. `Math.trunc(Math.random() * 1000000)`
+2. A NodeJS module like *UUID, shortid, nanoid*, etc.
+
+This tutorial uses the `nanoid` module to generate random numbers.
+
+1. In the terminal, install the `nanoid` module. This module will create a random set of characters as the OTP. Run the following command in your terminal:
+    
+    ```bash
+    npm i nanoid
+    ```
+    
+2. After installing the `nanoid` module, import it into `server.js` and use it to generate 6 random digits.
+    
+    ```js
+    const nanoid = require("nanoid");//Importing the naonid module
```

Comment:
@divofred  Got the following error after adding the `nanoid` module
![image](https://user-images.githubusercontent.com/16396865/176500348-a41331cc-3969-4371-adfd-6920dc7e29cb.png)

Reply:
@VershaGupta I'm sorry you got that error. I will fix it ASAP.

Reply:
@VershaGupta I've fixed the error. The latest version of nanoid doesn't support the use of `require` to import the module.
I specified the version of nanoid compatible with NodeJS.